src/pages/[platform]/build-a-backend/functions/examples/s3-upload-confirmation/index.mdx

```diff
@@ -0,0 +1,64 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'S3 Upload confirmation',
+  description:
+    'Use a trigger to confirm uploading files',
+  platforms: [
+    'android',
+    'angular',
+    'flutter',
+    'javascript',
+    'nextjs',
+    'react',
+    'react-native',
+    'swift',
+    'vue'
+  ]
+};
+
+export function getStaticPaths() {
+  return getCustomStaticPath(meta.platforms);
+}
+
+export function getStaticProps() {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+You can use `defineStorage` and `defineFunction` to create a function trigger to confirm uploading a file.
+
+First, update your storage definition to define the onUpload trigger as below:
+
+```ts title="amplify/storage/resource.ts"
+export const storage = defineStorage({
+  name: 'myProjectFiles',
+  triggers: {
+    onUpload: defineFunction({
+      entry: './on-upload-handler.ts'
+    })
+  }
+});
+```
+
+Next, create the function definitions at `amplify/storage/on-upload-handler.ts`.
+
+```ts title="amplify/storage/on-upload-handler.ts"
+import type { S3Handler } from 'aws-lambda';
+
+export const handler: S3Handler = async (event) => {
+  const objectKeys = event.Records.map((record) => record.s3.object.key);
+  console.log(`Upload handler invoked for objects [${objectKeys.join(', ')}]`);
+};
+```
+
+<Callout info>
+
+**Note:** The `S3Handler` type comes from the [`@types/aws-lambda`](https://www.npmjs.com/package/@types/aws-lambda) npm package. This package contains types for different kinds of Lambda handlers, events, and responses.
```

Comment:
instead of callout would it be better if we show the [aws-lambda](https://www.npmjs.com/package/@types/aws-lambda) installation as first step?